cst438_register/src/main/java/com/cst438/controller/CourseController.java

```diff
@@ -22,10 +24,22 @@
 	 */
 	@PutMapping("/course/{course_id}")
 	@Transactional
-	public void updateCourseGrades( @RequestBody CourseDTOG courseDTO, @PathVariable("course_id") int course_id) {
+	public void updateCourseGrades( @RequestBody CourseDTOG courseDTO, @PathVariable("course_id") int course_id) {	//courseDTOG, course_id
 		
 		//TODO  complete this method in homework 4
 		
+		CourseDTOG.GradeDTO gradeDTO = new CourseDTOG.GradeDTO();
+		
+		Enrollment e = enrollmentRepository.findByEmailAndCourseId(gradeDTO.student_email, courseDTO.course_id);
+		
+		if (e != null) {
+			e.setCourseGrade(gradeDTO.grade);
+			enrollmentRepository.save(e);
+		}
+		else {
+			throw  new ResponseStatusException( HttpStatus.BAD_REQUEST, "Could not find enrollment by email: " + gradeDTO.student_email );
```

Comment:
If there's an error, it could be that the course_id also does not exist, so this error might be misleading if that's the case.